### ISSUE TITLE
[FSR] Submission status metadata bug 

### ISSUE THREAD

**Issue by Dr-Pongo** [department-of-veterans-affairs/va.gov-team#72988]
## Description
Based on the in progress data stored in the BE it looks like the metadata submission status for 5655 isn't getting appropriately set in some* instances. It looks like the "has attempted submit" flag is being set just fine though. 

## Tasks
- [x] Investigate why submissions aren't triggering submission status to be appropriately set
- [ ] Fix issue with submissions status not being set

## Acceptance criteria

- [ ] Submission status for forms is being set predictably 

## Testing
This might rely heavily on platform functionality, so generating form specific test for this may be out of scope. so with that:
- [ ] (optional) Generate unit/e2e test to ensure consistent functionality

---

## Ticket creation

<details><summary>Ticket requirements</summary>

### Required to be "Ready"
- [x] Description written
- [x] Tasks defined
- [x] Acceptance criteria written
- [x] Testing defined
- [x] Add Labels
- [x] Attach to an Epic
- [ ] Estimate Points
</details>

**Comment by Dr-Pongo** [comment]
Still one point remaining as we start planning for sprint 120!

**Comment by Dr-Pongo** [comment]
Moved to blocked while I wait for some other eyes to look things over!

**Comment by Dr-Pongo** [comment]
Some notes from discovery!

So it's looking like we could use an additional form auto save on the confirmation page, or immediately after the `SET_SUBMISSION` reducer executes. The form submission metadata is being updated locally with statuses such as `serverError`, and `applicationSubmitted`, but after that status is set, there's no additional call to update the InProgressForm controller. 

Going to drop some file names here for future reference:
[SubmitController](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/review/SubmitController.jsx) (submit button on final review and submit page) utilizes [the submitForm action](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/actions.js) which  then calls `setSubmitted` if it's successful, and updates the submission status with application submitted or an error message if it fails. After that I can't seem to track another time the form auto saves, or updates the in progress controller. 

I spent some time investigating options, but my lack of familiarity with the platform forms-system as kind of left me at a dead end so I've reached out to some engineers with more experience. I'll likely need to create a platform ticket to hand this one off, or create a new ticket to implement the changes
